pipelines/placesapi/__init__.py

```diff
@@ -0,0 +1,94 @@
+"""
+ Uses the Google Places API to get detail information (Ratings, Reviews, Business Hours) for specified locations (Businesses / Public Places etc).
+ For more Info on Places API: "https://developers.google.com/maps/documentation/places/web-service/overview"  
+"""
+
+import dlt
+from dlt.sources.helpers import requests
+from dlt.extract.source import DltResource
+
+
+@dlt.source
+def places_api_source(api_secret_key=dlt.secrets.value) ->  DltResource:
+    """
+    The main source for dlt pipeline returns the text_search_resource.
+
+    @:params: api_secret_key - The api key for the Places API defined in dlt.secrets.
+    """
+    return text_search(api_secret_key)
+
+
+@dlt.resource(write_disposition="append")
+def text_search(api_secret_key=dlt.secrets.value):
+    """
+    Creates a text search resource. A Text Search returns information about a set of places based on a text string provided.
+
+    For more Info: "https://developers.google.com/maps/documentation/places/web-service/search-text"
+    
+    """
+
+    # API Endpoint.
+    text_search_url = "https://maps.googleapis.com/maps/api/place/textsearch/json?"
+
+    '''
+    The Parameters to be passsed to endpoint:
+    @:params: query - The text against which a search will be made.
+    @:params: radius  - Defines the distance (in meters) within which to return place results.
+    @:params: api_secret_key - The api key for the Places API defined in dlt.secrets.
+    '''
+    payload = {
+        'query' : 'FitX',
+        'radius' : '',
+        'key': {api_secret_key}
```

Comment:
The 'key' key in the payload dictionary is wrapped in curly braces which would make it a set in Python. If the API expects a string, remove the curly braces.

Reply:
Fixed.